### PR TITLE
fixed wind estimator

### PR DESCRIPTION
**Pull request by RomanLut**
1) groundVelocity[**X**] - lastGroundVelocity[**Z**]; is typo obviously

2) we want **fuselageDirection** and **groundVelocity** vectors to point in the same direction.

groundVelocity[Z] = gpsSol.velNED[Z] is POSITIVE when velocity vector points DOWN
(as can be seen in navigation_pos_estimator.c line 257: ```posEstimator.gps.vel.z = -gpsSol.velNED[Z];```).
Thus  we have to invert Z component:  ```fuselageDirection[Z] = -rMat[2][0];```

groundVelocity[Y] = gpsSol.velNED[Y] is POSITIVE when velocity vector points EAST (yaw = 90)
(as can be seen in gps.c line 292: ```gpsSol.velNED[Y] = speed * sin_approx(DECIDEGREES_TO_RADIANS(FAKE_GPS_GROUND_COURSE_DECIDEGREES));```).

rMat[1][0] is NEGATIVE at yaw=90, which can be found out by debugging, or by looking at imu.c line 461:

```attitude.values.yaw = RADIANS_TO_DECIDEGREES(-atan2_approx(rMat[1][0], rMat[0][0]));```

which equals to:

```attitude.values.yaw = RADIANS_TO_DECIDEGREES(atan2_approx(-rMat[1][0], rMat[0][0]));```

thus we have to negate rMat[1][0].

3) estimated wind vector is in NED frame because it is derived from gpsSol.velNED.
gpsSol.velNED[Z] is POSITIVE when velocity vector points DOWN
thus getEstimatedWindSpeed(Z) value should be negated in osd.c and position estimator.

 In other words, groundVelocity is in NED frame ( North, East, Down). The signs of rMat[x][0] can be checked by debugging, by pointing airplane north, east and down.

### CODE REVIEW

**Review by JulioCesarMatias — APPROVED** [review]
Based on the mathematical equations I reviewed from the [PDF file of William Premerlani](https://drive.google.com/file/d/0ByvTkVQo3tqXVzBYQUZicUNvbEE/view?resourcekey=0-s-7v3PgOUrZt7fI6JNlu-Q) (author of this wind estimation model for fixed-wing aircraft), I made sure that your PR is indeed correct, and corrects the Matrix Rotation problem. LGTM!

**Comment by RomanLut** [comment]
> But does it actually fix it so wind is estimated correctly/better than before ?

Existing implementation has obvious bugs, it does not work at all.

Anyway, I would like to hear someone else tested:

https://github.com/iNavFlight/inav/pull/8228 
and 
https://github.com/iNavFlight/inav/pull/8227

and confirm that wind estimator works very well with theese fixes.

**Comment by DzikuVx** [comment]
Let's merge and see what's going to happen

**Comment by b14ckyy** [comment]
Tested wind estimator today in HITL with simulated 24kph wind. During WP mission the wind estimation was within +/-2kph 
Even with multiple climbs and descents and some AHI drift in full sim mode. 
image

**Comment by Jetrell** [comment]
@RomanLut I've tested it a few times in real flight. With varying wind conditions, from a calm up to 25km/h.
I found the wind direction indicator to be a little more accurate... IMO it was always around 70% accurate, but now would appear to be in the 90% range... I personally find it more useful than the wind estimator.

While I used to find the wind speed estimator to be useless most of the time.
Now using a pitot as a reference... I would say its close to 90% accurate, when it has the best data to make the estimation.. But there are still plenty of times when it can't get it correct, based on the relative wind vector heading. 
Still, better than before. :+1:

**Comment by b14ckyy** [comment]
@Jetrell I think it is still most important during NAV mode flights. In this case even the old estimator worked really well. If it works even better now, that's great for longr ange pilots to better estimate remaining flight time and range. 

@RomanLut Does this new estimator still need the cruise speed (airspeed at crusie throttle in level flight) to work or is that not needed to be set for it to work? 
It's still needed for range calculation of course but curious if it is needed for future navigation tweaks.

**Comment by RomanLut** [comment]
1) Before #8228 and #8228 wind estimator did not work at all. You can revert those and see yourself in HITL.
2) cruise speed is not used by estimator. If you fly with a wind and then in oposite direction, it is possible to calculate wind speed as a half difference of ground speeds. Estimator does basically the same, but it estimates from difference of ground speeds when heading changes ~10 degrees. In order to get good estimation, do one full turn in cruse mode.
